lib/batch_loader/executor_proxy.rb

```diff
@@ -29,7 +30,11 @@ def load(item:, value:)
     end
 
     def loaded_value(item:)
-      loaded[item]
+      if value_loaded?(item: item)
```

Comment:
I just realized that with the changes the source code became not thread-safe :)

It's possible to have race-conditions when Thread1 read the `value_loaded?(item: item)` value and Thread2 loaded value after that, before Thread1 finished executing `loader`. Need to think how to avoid such situations, probably with some locking mechanisms (hope they won't affect performance a lot).

Reply:
Was able to reproduce the race-condition in https://github.com/exAspArk/batch-loader/pull/10. Will think what's the best way to fix it.